pkg/segment/structs/segstructs.go

```diff
@@ -452,9 +452,15 @@ type AggregationResult struct {
 	Results         []*BucketResult // histogram results
 }
 
+// TODO: Retain either IGroupByValues or GroupByValues, as having both is unnecessary.
+// The goal is to preserve the group-by value type as interface{} to avoid issues
+// when processing subsequent commands.
+// Ideally, we should update GroupByValues to have a type of []interface{}
+// and eliminate IGroupByValues.
 type BucketHolder struct {
-	GroupByValues []string
-	MeasureVal    map[string]interface{}
+	IGroupByValues []interface{} // each group-by value is stored as interface{}
```

Comment:
nit: maybe use `CValueEnclosure` or `DTypeEnclosure` instead of `interface{}`, so that it's more clear which types are valid